lib/mayaUsd/render/vp2RenderDelegate/renderDelegate.cpp

```diff
@@ -184,7 +194,9 @@ class MShaderCache final
             return;
 
         _3dDefaultMaterialShader
-            = shaderMgr->getStockShader(MHWRender::MShaderManager::k3dDefaultMaterialShader);
+            = TfGetEnvSetting(MAYAUSD_VP2_USE_STANDARDSURFACE_FALLBACK) ?
+            shaderMgr->getStockShader(MHWRender::MShaderManager::k3dStandardSurfaceShader) :
+            shaderMgr->getStockShader(MHWRender::MShaderManager::k3dDefaultMaterialShader);
```

Comment:
I would say lets live in the future and not use an environment variable in this situation. There is a MEL call that will tell you what the default shader name is:

```c++
    _3dDefaultMaterialShader
        = (MGlobal::executeCommandStringResult("defaultShaderName()") != "lambert1") ?
        shaderMgr->getStockShader(MHWRender::MShaderManager::k3dStandardSurfaceShader) :
        shaderMgr->getStockShader(MHWRender::MShaderManager::k3dDefaultMaterialShader);
```

If you are curious or want to test that everything still works in a legacy universe, you can either open an old scene that was saved when Lambert was the default, or use the following env var to temporarily restore Lambert as the default: `set MAYA_DEFAULT_SURFACE_SHADER=lambert`.

Reply:
Oh. That brings back bad memories of PR #2049. The number of unit test results to update would be quite important if you modernize the code. 

That would be asking a lot, so let's try a progressive solution: introduce an env var, but instead of having it introduce the new behavior, use it to get back to the previous one:
```c++
    bool useStandardSurface = (MGlobal::executeCommandStringResult("defaultShaderName()") != "lambert1");
    if (TfGetEnvSetting(MAYAUSD_VP2_USE_LAMBERT_FALLBACK) {
        // Explicit request for legacy Lambert:
        useStandardSurface  = false;
    }
    _3dDefaultMaterialShader
        = useStandardSurface ?
        shaderMgr->getStockShader(MHWRender::MShaderManager::k3dStandardSurfaceShader) :
        shaderMgr->getStockShader(MHWRender::MShaderManager::k3dDefaultMaterialShader);
```

This way you can add that env var to `test\lib\mayaUsd\render\vp2RenderDelegate\CMakeLists.txt` to not have to update all results, and remove the env var for unit tests that would shine with your new code.

One unit test I have in mind would be `test\lib\mayaUsd\render\vp2RenderDelegate\testVP2RenderDelegateDisplayColors.py` with a standardSurface-based scene when `mel.eval("defaultShaderName()") != "lambert1"`. Would require a different set of baseline images in the modern code path. Just add a _standardSurface suffix to them like we [do in other modernization situations](https://github.com/Autodesk/maya-usd/blob/dev/test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateMaterialX.py#L100).

Reply:
I think it's a great idea to live in the future and have the env var work the other way around. Thanks for the hint of the mel command to get the default shader name, this is a good way to go.

Reply:
- Changed to `MAYAUSD_VP2_USE_LAMBERT_FALLBACK` and comparison of the default shader name.
- Updated tests cmake file to use this environment for legacy tests.